internal/component/prometheus/operator/common/crdmanager.go

```diff
@@ -469,7 +470,11 @@ func (c *crdManager) addServiceMonitor(sm *promopv1.ServiceMonitor) {
 	mapKeys := []string{}
 	for i, ep := range sm.Spec.Endpoints {
 		var scrapeConfig *config.ScrapeConfig
-		scrapeConfig, err = gen.GenerateServiceMonitorConfig(sm, ep, i)
+		role := promk8s.Role(c.args.KubernetesRole)
+		if role == "" {
+			role = promk8s.RoleEndpoint
+		}
+		scrapeConfig, err = gen.GenerateServiceMonitorConfig(sm, ep, i, role)
```

Comment:
```suggestion
		scrapeConfig, err = gen.GenerateServiceMonitorConfig(sm, ep, i, promk8s.Role(c.args.KubernetesRole))
```
We can probably do without the `role` variable.